src/datasources/db/v2/postgres-database.service.spec.ts

```diff
@@ -0,0 +1,169 @@
+import { join } from 'path';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Test, type TestingModule } from '@nestjs/testing';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { postgresConfig } from '@/config/entities/postgres.config';
+import { ConfigurationModule } from '@/config/configuration.module';
+import configuration from '@/config/entities/__tests__/configuration';
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { PostgresDatabaseService } from '@/datasources/db/v2/postgres-database.service';
+import { DataSource, type ObjectLiteral, type Repository } from 'typeorm';
+import { DatabaseMigrator } from '@/datasources/db/v2/database-migrator.service';
+import { DatabaseShutdownHook } from '@/datasources/db/v2/database-shutdown.hook';
+import { DatabaseInitializeHook } from '@/datasources/db/v2/database-initialize.hook';
+
+describe('PostgresDatabaseService', () => {
+  let moduleRef: TestingModule;
+  let postgresqlService: PostgresDatabaseService;
+
+  beforeAll(async () => {
+    // We should not require an SSL connection if using the database provided
+    // by GitHub actions
+    const isCIContext = process.env.CI?.toLowerCase() === 'true';
+    const baseConfiguration = configuration();
+    const testConfiguration: typeof configuration = () => ({
+      ...baseConfiguration,
+      db: {
+        ...baseConfiguration.db,
+        connection: {
+          postgres: {
+            ...baseConfiguration.db.connection.postgres,
+            ssl: {
+              ...baseConfiguration.db.connection.postgres.ssl,
+              enabled: !isCIContext,
+              caPath: join(__dirname, '../../../../db_config/test/server.crt'),
+            },
+          },
+        },
+      },
+    });
+
+    moduleRef = await Test.createTestingModule({
+      imports: [
+        TypeOrmModule.forRootAsync({
+          imports: [ConfigModule],
+          useFactory: async (configService: ConfigService) => {
+            const typeormConfig = await configService.getOrThrow('db.orm');
+            const postgresConfigObject = postgresConfig(
+              await configService.getOrThrow('db.connection.postgres'),
+            );
+
+            return {
+              ...typeormConfig,
+              ...postgresConfigObject,
+            };
+          },
+          inject: [ConfigService],
+        }),
+        TestLoggingModule,
+        ConfigurationModule.register(testConfiguration),
+        ConfigModule,
+      ],
+      providers: [
+        DatabaseMigrator,
+        DatabaseShutdownHook,
+        DatabaseInitializeHook,
+        PostgresDatabaseService,
+      ],
+    }).compile();
+
+    postgresqlService = moduleRef.get<PostgresDatabaseService>(
+      PostgresDatabaseService,
+    );
+  });
+
+  afterAll(async () => {
+    await postgresqlService.destroyDatabaseConnection();
+    await moduleRef.close();
+  });
```

Comment:
Nit: similar to your comment in the other PR, I would `beforeEach`/`afterEach` this. The tests are then decoupled from each other and we can be sure that the database is always initialised at the start of a test.

For example, when you test the `isInitialised` method, you are reinitialising the database as it is required in the subsequent test.

Reply:
Although I believe we shouldn't create a module every time, I made this change for now to maintain consistency.
https://github.com/safe-global/safe-client-gateway/pull/1988/commits/a29425089cb674ccc6dc44526ab72e538d4a7968

Reply:
It wasn't with regards to the module itself, but the database state - ensuring that there was no "bleed" between tests. Before, the initialisation state was dependant on previous tests. Now, tests always start with the database not initialised.

However, it was a nit. If you have a strong opinion, you can go with your preference.

Reply:
All good, Let's keep it in `before/afterEach` as you suggested for now.